Add GOVERNANCE.md, MAINTAINERS.md and expand contribution paths (#1102)
valkey-io/valkey-search: GOVERNANCE.md

# Project Governance

Valkey Search is a sub-project of the [Valkey](https://github.com/valkey-io/valkey) project.
It aspires to the same style of governance as the core Valkey project, whose model is described in the core [GOVERNANCE.md](https://github.com/valkey-io/valkey/blob/unstable/GOVERNANCE.md).
Contributors familiar with the core project should find the roles, decision-making, and expectations here intentionally similar.

## Relationship to the Valkey project

Because Valkey Search is a sub-project under the Valkey-io organization, the Valkey Technical Steering Committee (TSC) holds ultimate authority over it.

> The TSC may, at its discretion, overrule decisions made within this project, although — consistent with the core governance document — it shall show restraint in doing so.

Reviewer: It's not wrong, but maybe it can be rephrased better:

```suggestion
Valkey Search is a sub-project within the Valkey project which is managed by the Valkey Technical Steering Committee (TSC).
The TSC has delegated management of this subproject to the provided list of maintainers. The TSC reserves the ability to overrule decisions made within this project, although they shall show restraint in doing so.
```

WDYT?

Our explicit goal is for Valkey Search to be **self-governing and self-contained**: day-to-day technical and project decisions should be made by this project's own maintainers, without requiring TSC involvement.
TSC interaction should be the exception (for example, escalation of an unresolved dispute or a change in the project's relationship to the broader Valkey organization), not part of the normal flow of development.

## Roles

* **Committers** are individuals with write access to the code within this repository.

> * **Maintainers** are individuals with full access to the repository who own its governance.

Reviewer: ```suggestion
* **Maintainers** are individuals with full access to the repository and own the project's management.
```

Technically the governance is the Valkey project governance, but the management has been delegated.
